### PR TITLE
papermill-like API

### PR DESCRIPTION
**Pull request by edublancas**
This PR adds a bunch of things to make it easier for papermill users to migrate to ploomber-engine

- a CLI (same arguments as papermill
- Python API (same arguments as papermill
- refactored docs using the new papermill-like API
- improved API documentation section
- also fixes a bug that was breaking notebook execution when notebooks had markdown cells

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -36,7 +36,9 @@ jobs:
 
       - name: Unit tests
         run: |
-          pytest --ignore=tests/test_engine.py
+          # NOTE: we need to pass tests/ otherwise pytest will load
```
Comment (idomic) [inline .github/workflows/ci.yml:39]: This is weird, how come it didn't happen so far?

File: doc/quick-start.md [hunk 2]
```diff
@@ -30,41 +29,56 @@ or
...
 
-Store the notebook:
+```{code-cell} ipython3
+_ = execute_notebook("nb.ipynb", "output.ipynb", profile_memory=True)
+```
 
-```python
-nbformat.write(out, "output.ipynb")
+```{code-cell} ipython3
+ls output-*
```
Comment (idomic) [inline doc/quick-start.md:83]: missing `!` ?
or `{code-cell} shell`?

File: doc/user-guide/profiling/memory.md [hunk 3]
```diff
@@ -13,26 +13,25 @@ kernelspec:
...
-```{note}
-You can download this tutorial in Jupyter notebook format by clicking on the icon in the upper right corner.
+```{versionadded} 0.0.18
+[`execute_notebook`](../../api/api.html#execute-notebook)
 ```
 
-```{important}
-Memory profiling requires ploomber-engine `0.0.14` or higher
-```
+With ploomber-engine you can profile Jupyter notebook's memory usage. Unlike papermill, which isn't capable of doing it.
```
Comment (idomic) [inline doc/user-guide/profiling/memory.md:21]: If they have an open issue about it, let's tag users and respond there.

File: doc/user-guide/running.md [hunk 4]
```diff
@@ -28,24 +34,46 @@ Download sample notebook:
...
-from ploomber_engine.ipython import PloomberClient
+from ploomber_engine import execute_notebook
 
-client = PloomberClient.from_path("running-demo.ipynb")
+nb = execute_notebook("running-demo.ipynb",
+                      output_path="output.ipynb")
 ```
 
-Execute the notebook:
+The function returns a notebook object (same contents as the stored in `output_path`):
```
Comment (idomic) [inline doc/user-guide/running.md:52]: The function returns a notebook object (same contents as ~~the~~ stored in `output_path`):

File: src/ploomber_engine/ipython.py [hunk 5]
```diff
@@ -165,24 +165,31 @@ def __init__(self, nb, display_stdout=False):
...
         """Initialize client from a path to a notebook
 
-
         Parameters
         ----------
         path : str
             Path to the ``.ipynb`` file
 
+        display_output : bool, default=False
+            If True, it prints whatever output the notebook prints.
```
Comment (idomic) [inline src/ploomber_engine/ipython.py:177]: If True, it prints whatever output the notebook prints. ->
If True, it prints the same output the notebook prints.

File: tests/test_execute.py [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+    )
+
+    execute_notebook(nb_in, "out.ipynb", log_output=True)
+
+    captured = capsys.readouterr()
+    assert captured.out == "hello\n"
+    assert captured.err == ""
+
+
+@pytest.mark.xfail(reason="not implemented")
```
Comment (idomic) [inline tests/test_execute.py:102]: Either implement or remove?

File: tests/test_execute.py [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+    nb_in = _make_nb(["import sys", "print('hello', file=sys.stderr)"])
+
+    execute_notebook(nb_in, "out.ipynb", log_output=True)
+
+    captured = capsys.readouterr()
+    assert captured.out == ""
+    assert captured.err == "hello\n"
+
+
+# TODO: we should check the contents of the plot
```
Comment (idomic) [inline tests/test_execute.py:113]: Either in or out, no pending TODOs

File: tests/test_execute.py [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+def test_execute_notebook_profile_runtime(cells, tmp_empty):
+    nb_in = _make_nb(cells)
+
+    execute_notebook(nb_in, "out.ipynb", profile_runtime=True)
+
+    assert Path("out-runtime.png")
+    assert Image(filename="out-runtime.png")
+
+
+# TODO: we should check the contents of the plot
```
Comment (idomic) [inline tests/test_execute.py:131]: Same as above

**Review by idomic — CHANGES_REQUESTED** [review]
Added some comments, please add the missing tests

File: .github/workflows/ci.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (edublancas) [reply .github/workflows/ci.yml:39]: cause I just added a new `src/conftest.py` in this PR

File: doc/_config.yml [hunk 9]
```diff
@@ -4,6 +4,7 @@
 title: ploomber-engine
 author: Ploomber
 logo: square-no-bg-small.png
+copyright: "2023"
```
[the remark this answers is not in the document]
Comment (edublancas) [reply doc/_config.yml:7]: in doc website I've seen they only put the [current year](https://jupyterbook.org/en/stable/intro.html), the -present thing I've only seen it in LICENSE files

File: doc/quick-start.md (hunk 2 above)
Comment (edublancas) [reply doc/quick-start.md:83]: no, that works. Jupyter recognizes `ls`

File: doc/user-guide/running.md (hunk 4 above)
Comment (edublancas) [reply doc/user-guide/running.md:52]: fixed

File: src/ploomber_engine/ipython.py (hunk 5 above)
Comment (edublancas) [reply src/ploomber_engine/ipython.py:177]: fixed

File: .github/workflows/ci.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (idomic) [reply .github/workflows/ci.yml:39]: I mean in other repos. It's usually in ./tests/ why here it's different?

File: doc/_config.yml (hunk 9 above)
[the remark this answers is not in the document]
Comment (idomic) [reply doc/_config.yml:7]: Alright.

File: tests/test_execute.py (hunk 6 above)
Comment (edublancas) [reply tests/test_execute.py:102]: I opened this [issue](https://github.com/ploomber/ploomber-engine/issues/24) to document what features we're missing. I think it's best to leave it there so whoever implements it doesn't have to implement it (just remove the xfail decorator)

Comment (idomic) [reply tests/test_execute.py:102]: Fair ok. In general we should avoid pushing todos/code with missing tests.

File: tests/test_execute.py (hunk 7 above)
Comment (edublancas) [reply tests/test_execute.py:113]: removed TODO, opened issue https://github.com/ploomber/sklearn-evaluation/issues/170

File: tests/test_execute.py (hunk 8 above)
Comment (edublancas) [reply tests/test_execute.py:131]: removed TODO, opened issue https://github.com/ploomber/sklearn-evaluation/issues/170

File: .github/workflows/ci.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (edublancas) [reply .github/workflows/ci.yml:39]: discussed on Slack

**Pull request merged**